### PR TITLE
[codestyle] More efficient way to convert char to string

### PR DESCRIPTION
**Pull request by ApCoder123**
Resolves issue here: http://sonar.xwiki.org/issues/search#issues=0d5c6230-4a67-4976-bd15-ef9db137c3b5

### CODE REVIEW

File: xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java [hunk 1]
```diff
@@ -102,7 +102,7 @@
     /**
      * Replacements chars for the escapes to be removed from the interwiki content.
      */
-    private static final String[] ESCAPE_REPLACEMENTS_INTERWIKI = new String[] { "" + ESCAPE_CHAR, "" };
+    private static final String[] ESCAPE_REPLACEMENTS_INTERWIKI = new String[] { String.valueOf(ESCAPE_CHAR) };
```
Comment (tmortagne) [inline xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java]: Actually you removed one too many `""`.

**Comment by ApCoder123** [comment]
I fixed the error

File: xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java [hunk 2]
```diff
@@ -102,7 +102,7 @@
     /**
      * Replacements chars for the escapes to be removed from the interwiki content.
      */
-    private static final String[] ESCAPE_REPLACEMENTS_INTERWIKI = new String[] { "" + ESCAPE_CHAR, "" };
+    private static final String[] ESCAPE_REPLACEMENTS_INTERWIKI = new String[] { String.valueOf(ESCAPE_CHAR), " " };
```
Comment (tmortagne) [inline xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java]: It's supposed to be an empty string, you modified it to a String containing a white space.

**Comment by tmortagne** [comment]
Ha looks like someone else fixed it.

**Pull request closed without merging**